src/coreclr/src/jit/hwintrinsicxarch.cpp

```diff
@@ -1173,19 +1173,209 @@ GenTree* Compiler::impBaseIntrinsic(NamedIntrinsic        intrinsic,
 
 GenTree* Compiler::impSSEIntrinsic(NamedIntrinsic intrinsic, CORINFO_METHOD_HANDLE method, CORINFO_SIG_INFO* sig)
 {
-    GenTree* retNode  = nullptr;
-    GenTree* op1      = nullptr;
-    GenTree* op2      = nullptr;
-    GenTree* op3      = nullptr;
-    GenTree* op4      = nullptr;
-    int      simdSize = HWIntrinsicInfo::lookupSimdSize(this, intrinsic, sig);
+    GenTree*  retNode  = nullptr;
+    GenTree*  op1      = nullptr;
+    GenTree*  op2      = nullptr;
+    int       simdSize = HWIntrinsicInfo::lookupSimdSize(this, intrinsic, sig);
+    var_types baseType = TYP_UNKNOWN;
 
     // The Prefetch and StoreFence intrinsics don't take any SIMD operands
     // and have a simdSize of 0
     assert((simdSize == 16) || (simdSize == 0));
 
     switch (intrinsic)
     {
+        case NI_SSE_CompareGreaterThan:
```

Comment:
Since these cases are all basically the same, with different constants, it would seem that you could add a function to determine the constant to use for the AVX intrinsic, based on the constant associated with the SSE intrinsic. Does that make sense or am I missing something?

Reply:
Yes, we probably could define a `getInverseFloatingComparison` method or something similar. A similar function could also be useful in lowering as it would allow either operand to be contained.

Reply:
I added a new function `HWIntrinsicInfo::lookupFloatingComparisonForSwappedArgs` and cleaned up the importation logic.